docs/DEVELOPING.md

```diff
@@ -737,6 +737,139 @@ $ make ftest NAME=mssql
 
 ℹ️ The e2e test uses default values defined in [Configure Microsoft SQL connector](#configure-microsoft-sql-connector)
 
+
+## SharePoint Connector
+
+The [Elastic SharePoint connector](../connectors/sources/sharepoint.py) is provided in the Elastic connectors python framework and can be used via [build a connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
+
+### Availability and prerequisites
+
+⚠️ _Currently, this connector is available in **beta** starting in 8.8_.
+Features in beta are subject to change and are not covered by the service level agreement (SLA) of features that have reached general availability (GA).
+
+Elastic versions 8.6.0+ are compatible with Elastic connector frameworks. Your deployment must include the Elasticsearch, Kibana, and Enterprise Search services.
+
+SharePoint Online permissions required to run the connector:
+     
+   - Refer the document to understand the complete process of setting [SharePoint Permissions](https://documentation.commvault.com/v11/essential/128616_registering_azure_app_for_sharepoint_online.html)
+
+   - To understand how to set DisableCustomAppAuthentication refer the document: [SharePoint Set DisableCustomAppAuthentication](https://learn.microsoft.com/en-us/answers/questions/714147/token-type-is-not-allowed-error-on-sharepoint-rest#:~:text=Install%2DModule%20%2DName,spotenant%20%2DDisableCustomAppAuthentication%20%24false)
+
+
+### Setup and basic usage
+
+Complete the following steps to deploy the connector:
+
+1. [Gather SharePoint details](#gather-sharepoint-details)
+2. [Configure SharePoint connector](#configure-sharepoint-connector)
+
+#### Gather SharePoint details
+
+Collect the information that is required to connect to your SharePoint instance:
+
+- The server host url where the SharePoint is hosted
+- Username for SharePoint Server
+- Password for SharePoint Server
+- Client Id for SharePoint Online
+- Secret Id for SharePoint Online
+- Tenant Name for SharePoint Online
+- Tenant Id for SharePoint Online
+- SSL certificate for a secure connection
+
+#### Configure SharePoint connector
+
+The following configuration fields need to be provided for setting up the connector:
+
+##### `is_cloud`
+
+Flag to determine the SharePoint platform type. `True` if SharePoint cloud and `False` if SharePoint Server. Default value is `False`.
```

Comment:
Should we use 'on premise' instead of 'server'?

Reply:
we have used  SharePoint server  instead of on premise because official documents refers it  with**Server**, But if you want we can change this.
https://learn.microsoft.com/en-us/sharepoint/sharepoint-server

Reply:
No worries, sounds good to me 👍